### DISCUSSION TITLE
Feature Request to Extend the Functionality of combine_split_data

### DISCUSSION THREAD

**Discussion by OCsteve** [timeseriesAI/tsai#226] [Ideas]
It would be very helpful to extend the functionality of combine_split_data() from 2-way to 3-way for numpy arrays that have already been appropriately created.

X, y, splits = combine_split_data([Xtrain, Xval], [Ytrain, Yval])
X, y, splits = combine_split_data([Xtrain, Xval, Xtest], [Ytrain, Yval, Ytest])

For use such as:
probas_val, _, preds_val = learn.get_X_preds(X[splits[1]])
probas_test, _, preds_test = learn.get_X_preds(X[splits[2]])

**Comment by oguiza** [comment]
Hi @OCsteve,
There's no need to modify anything. That's exactly how tsai works. You can test it using this code:

```bash
from tsai.all import *
X_train, y_train, X_valid, y_valid = get_UCR_data("LSST")
X, y, splits = combine_split_data([X_train, X_valid, X_train], [y_train, y_valid, y_train])
len(splits), np.array_equal(X_train, X[splits[2]])
```

**Reply by OCsteve** [reply]
That code snippet works. It appears the error I'm getting is from the next line of my code that calls check_data().

X_train, y_train, X_valid, y_valid = get_UCR_data("LSST")
print(X_train.shape, y_train.shape, X_valid.shape, y_valid.shape)

X, y, splits = combine_split_data([X_train, X_valid, X_train], [y_train, y_valid, y_train])
len(splits), np.array_equal(X_train, X[splits[2]])

check_data(X, y, splits) # TypeError: check_overlap() takes 2 positional arguments but 3 were given

# ===
# Now try multivariate y's:
Ty = 10
y_train = np.ones([len(y_train), Ty]).astype('float32')  
y_valid = np.ones([len(y_valid), Ty]).astype('float32')
print(X_train.shape, y_train.shape, X_valid.shape, y_valid.shape)

X, y, splits = combine_split_data([X_train, X_valid, X_train], [y_train, y_valid, y_train])
len(splits), np.array_equal(X_train, X[splits[2]])

check_data(X, y, splits) # TypeError: check_overlap() takes 2 positional arguments but 3 were given

**Comment by oguiza** [comment]
I’ve created an issue and added the requested functionality to tsai.
